numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -2459,46 +2458,49 @@ einsum_list_to_subscripts(PyObject *obj, char *subscripts, int subsize)
             subscripts[subindex++] = '.';
             ellipsis = 1;
         }
-        /* Subscript */
-        else if (PyInt_Check(item) || PyLong_Check(item)) {
-            long s = PyInt_AsLong(item);
-            npy_bool bad_input = 0;
+        else {
+            npy_intp s = PyArray_PyIntAsIntp(item);
+            /* Subscript */
+            if (!PyErr_Occurred()) {
+                npy_bool bad_input = 0;
 
-            if (subindex + 1 >= subsize) {
-                PyErr_SetString(PyExc_ValueError,
-                        "subscripts list is too long");
-                Py_DECREF(obj);
-                return -1;
-            }
+                if (subindex + 1 >= subsize) {
+                    PyErr_SetString(PyExc_ValueError,
+                            "subscripts list is too long");
+                    Py_DECREF(obj);
+                    return -1;
+                }
 
-            if ( s < 0 ) {
-                bad_input = 1;
-            }
-            else if (s < 26) {
-                subscripts[subindex++] = 'A' + (char)s;
-            }
-            else if (s < 2*26) {
-                subscripts[subindex++] = 'a' + (char)s - 26;
-            }
-            else {
-                bad_input = 1;
-            }
+                if ( s < 0 ) {
```

Comment:
```suggestion
                if (s < 0) {
```
Ah, was there before... but I think I will commit it anyway...